File: libsrr/example_client.py
# Copyright 2020 David Lareau. This program is free software under the terms of the GPL-3.0-or-later, no warranty.
# LD_LIBRARY_PATH=. python -B example_client.py

from srr import srr

with srr('/example-srr') as client:
  reply = client.send("hello".encode())
  print(len(reply))
  print(reply)
  print(bytearray(reply).decode())

File: libsrr/example_client.java
// Copyright 2020 David Lareau. This program is free software under the terms of the GPL-3.0-or-later, no warranty.
// javac -classpath .. example_client.java
// LD_LIBRARY_PATH=. java -cp .:.. -Djava.library.path=$(pwd) example_client

import libsrr.*;

class example_client {

  public static void main(String [] args) {
    try(srr srr = new srr("/example-srr", 8192, false, false, 3)) {
      System.out.println(srr.as_string(srr.send("hello")));
    } catch(Exception e) { e.printStackTrace(); }
  }

}
